src/utils/saga/paged.js

```diff
@@ -0,0 +1,123 @@
+import { call } from 'redux-saga/effects';
+
+import request from 'utils/request';
+
+/**
+ * Counter Helper
+ * @param {url} url Route to the counter
+ * @param {type} type What is being counted
+ * @eample { url: '/roles/count', type: 'Roles' }
+ */
+export function* counter({ url, type }) {
+  const endpoint = {
+    url,
+    method: 'GET'
+  };
+  const count = yield call(request.execute, { endpoint });
+
+  if (count.success) {
+    const {
+      response: {
+        data: { result }
+      }
+    } = count;
+
+    return result;
+  } else if (count.error) {
+    throw count.error;
+  } else {
+    throw new Error(`Failed to count ${type}!`);
+  }
+}
+
+/**
+ * Pager Helper
+ * @param {req} req
+ * @example { cached: rolesCached,
+  pager: {
+    ...pager,
+    store: rolesPager
+  },
+  path: {
+    counter: '/roles/count',
+    query: '/roles/'
+  },
+  type: 'Roles' }
+ */
+export function* pager({ req }) {
+  const Pager = {};
```

Comment:
This function should return a class instance rather than building up an object with properties.

Reply:
I restructured this, so pager and paged are just a single function - pager. I couldn't think of an instance when they were would needed separately.